### PR TITLE
Add filtering export

### PR DESCRIPTION
**Pull request by sedroche**
Adding filtering to the export feature.
Mimicking current functionality as they may be side by side if only import/export released first.

- Added a middleware to format current ngui filter object into a mongo query
- New tests
- Updated docs

### CODE REVIEW

File: src/middleware/mongoQuery/index.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Ensure that passed values are of the type indicated.
+ */
+var casters = {
+  'String': function(val) {
+    return `${val}`;
+  },
+  'ObjectId': function(val) {
+    if (mongodb.ObjectId.isValid(val)) {
+      return mongodb.ObjectId(val);
```
Comment (pb82) [inline src/middleware/mongoQuery/index.js]: According to http://mongodb.github.io/node-mongodb-native/2.1/api/ObjectID.html you should use `new` to create a new ObjectId

Comment (sedroche) [reply src/middleware/mongoQuery/index.js]: :+1:

**Review by rachael-oregan — APPROVED** [review]
LGTM

File: src/middleware/mongoQuery/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Middleware for parsing filter requests into mongo db queries.
+ *
+ */
+export default function() {
+
+  return (req, res, next) => {
+    if (!req.query.filter) {
+      req.query.mongoQuery = {};
+      return next(null);
```
Comment (pb82) [inline src/middleware/mongoQuery/index.js]: why `next(null)` instead of just `next()`?

**Review by pb82 — APPROVED** [review]
LGTM, just two comments

File: src/middleware/mongoQuery/index.js (hunk 2 above)
Comment (sedroche) [reply src/middleware/mongoQuery/index.js]: Not sure why I did that. Will remove it, cheers

**Pull request merged**